### PR TITLE
MATEO-3

### PR DESCRIPTION
**Pull request by mateomustapic**
main.js
- defined function as a property of portlet instance js object
- used instance.byId to avoid DOM global searches
- removed unnecessary if check in countTasks function
- implemented localization using UnicodeLanguageUtil() in jsp

view.jsp
- used portlet namespace for elements used in javasript
- created tab for profile page
- included contact.jsp
- implemented localization for taskFinished label in javascript using UnicodeLanguageUtil()

### CODE REVIEW

File: mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/view.jsp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<%--
+<%@ page import="com.liferay.portal.kernel.language.UnicodeLanguageUtil" %><%--
```
Comment (markocikos) [inline mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/view.jsp:1]: All imports should be in init.jsp.

File: mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/view.jsp [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
 <div class="list-container">
 	<div class="header">
+
```
Comment (markocikos) [inline mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/view.jsp:21]: Two base rules of formatting HTML, as mentioned before:
1. Child node should indented by tab, no blank row.
1. Sibling nodes should be separated by a blank row.

It was fine as it was.

File: mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/view.jsp [hunk 3]
```diff
@@ -53,74 +58,64 @@
...
 				<h3>
-					<liferay-ui:message key="finished-tasks" />
+
+					<liferay-ui:message key="contact-us" />
+
+					<i class="icon-envelope"></i>
 				</h3>
 
-				<ul class="finished-task-list"></ul>
+				<%@ include file="contact.jsp" %>
```
Comment (markocikos) [inline mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/view.jsp:118]: You are including file that doesn't exist. There is only error in UI.

If you are including a JSP, the path should start with ```/``` and it should be a JSP**F** (JSP Fragment), so Intellij knows how to parse it.

File: mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/view.jsp [hunk 4]
```diff
@@ -129,7 +124,8 @@
 <aui:script use="todo">
 	new Liferay.Portlet.Todo(
 		{
-			namespace: '<portlet:namespace />'
+			namespace: '<portlet:namespace />',
+			taskFinished: '<%= UnicodeLanguageUtil.get(pageContext, "task-finished") %>'
```
Comment (markocikos) [inline mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/view.jsp:128]: You still did not include ```Language.properties``` file, so localization still does not work. 

I see you sent it in a separate PR a day later, but one PR should be one functional whole, unless you have a very good reason why you are sending only a part, like too much code. Language.properties does not fall into that category, as it does not contain any code.

File: mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/js/main.js [hunk 5]
```diff
@@ -20,7 +28,32 @@ AUI.add(
...
+
+						var todoList = instance.byId('todo-list');
+
+						instance._todoList = todoList;
+
+						var finishedTask = instance.byId('finished-task');
+
+						var taskInput = instance.byId('task');
+
+						var taskListLI = todoList.all('li');
```
Comment (markocikos) [inline mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/js/main.js:46]: Use camelCase for variable names. Also,```Li``` suffix is clunky and does represent meaning of the content, something like ```taskListItems``` would make more sense.

File: mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/js/main.js [hunk 6]
```diff
@@ -20,7 +28,32 @@ AUI.add(
...
+
+						instance._tasksListLi = taskListLI;
```
Comment (markocikos) [inline mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/js/main.js:48]: ```instance._tasksListLi``` is not used anywhere.

File: mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/js/main.js [hunk 7]
```diff
@@ -70,7 +149,27 @@ AUI.add(
 							todoList.append(taskHtml);
 
 							taskInput.val('');
+
+							instance._countTasks();
 						}
+					},
+
+					_countTasks: function() {
```
Comment (markocikos) [inline mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/js/main.js:157]: The function does more then counts tasks, it also updates UI. it should be named something like "_updateTaskCounts".

File: mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/js/main.js [hunk 8]
```diff
@@ -70,7 +149,27 @@ AUI.add(
...
+						var instance = this;
+
+						var todoList = instance._todoList;
+
+						var taskListLi = todoList.all('li');
+
+						var remainingCount = taskListLi.size();
+
+						var remainingDiv = instance.byId('remaining-tasks-count');
```
Comment (markocikos) [inline mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/js/main.js:166]: You are making unnecessary DOM queries within a click event handler. As I mentioned before, if the queried elements never change you should query then only once in ```initializer()``` and store it as ```instance``` property. See ```instance._appendTodoList()```. The only case when DOM queries from event handler make sense is if you are querying element relative to click location, like on line 111.

**Comment by markocikos** [comment]
Not merged.

Please make the changes in comments and resend PR.

Thank you!

/cc @ivicac @igorbeslic

File: mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/js/main.js (hunk 7 above)
Comment (igorbeslic) [reply mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/js/main.js:157]: I would just add that method/function should do only what its name says it does. Such naming and strict rules will make code readable and easier for maintenance for other developers.
http://www.goodreads.com/book/show/3735293-clean-code

**Pull request closed without merging**